### ISSUE TITLE
Bug: [consistent-type-imports] reports "used by decorator metadata" if module has identically named value.

### ISSUE THREAD

**Issue by simonbuchan** [typescript-eslint/typescript-eslint#7101]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.0.4&sourceType=module&code=JYWwDg9gTgLgBDAnmApnA3nAYhCcC+cAZlBCHAEQB0A9EbhQNwCwAUGwMYQB2AzvDjwBeOAEYATAGYWrIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyo0kRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
import type { Foo } from "./foo";

const Foo = 123;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/consistent-type-imports": "error"
  }
};
```

### tsconfig

```jsonc
{
}
```

### Expected Result

No error reported.

### Actual Result

Reports: `Type import "Foo" is used by decorator metadata. 1:1 - 1:34`

### Additional Info

It might seem a bit weird to do this, but it can occasionally come up. My use case was importing a type that I was validating locally with `zod` (a runtime type validation library), where I have elsewhere consistently used the convention:

```ts
export const SomeType = z.object({ foo: z.string() });
export type SomeType = z.infer<SomeType>;
```

This lets you use just `import { SomeType } ...` and use it both as a type and as a validator for said type. Here I wanted to validate an external type, so I declared a local `zod` schema matching the type as usual, verifying it matched the type:

```ts
import type { SomeType } ...;

const SomeType: z.Schema<SomeType> = ...;
```

which causes the issue.

The obvious workaround is to simply use a different name.

**Comment by Josh-Cena** [comment]
This error message no longer exists
